ci/cppcheck/cppcheck-new-issues.sh

```diff
@@ -25,8 +25,19 @@ do
     if ! grep -q -F "$i" "$rootdir"/ci/cppcheck/cppcheck_existing_issues.txt ; then
         status=1
         err "New issue:"
-        grep -F "$i" "$rootdir/cppcheck_results.txt"
+        "$i"
     fi
 done
 
+# Generate a new list of existing issues by removing the ones that are now solved
+comm -12 <(sort "$rootdir"/cppcheck_results.txt) <(sort "$rootdir"/ci/cppcheck/cppcheck_existing_issues.txt)  > "$rootdir"/cppcheck_existing_issues.txt
+
+mv "$rootdir"/cppcheck_existing_issues.txt "$rootdir"/ci/cppcheck/cppcheck_existing_issues.txt
+
+if ! git diff -U0 --exit-code -- "$rootdir"/ci/cppcheck/cppcheck_existing_issues.txt ; then
```

Comment:
I think this should only be done if status was still 0. Otherwise, you'll get both the report of the new issue above, and again in the diff here.